doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,291 @@ Reasoning with inductive types
+     .. note::
+        To destruct a number, use the syntax :n:`destruct (@natural)` (not
+        very interesting anyway).
 
-      This tactic behaves like :n:`destruct @term` except that it does not
+   :n:`@ident` (in :n:`@destruction_arg`)
+     If :token:`ident` is a quantified variable in the conclusion
+     of the goal, then :n:`destruct @ident` behaves like :tacn:`intros`
+     :n:`until @ident; destruct @ident`.
+     If :token:`ident` is a hypothesis of the context and :token:`ident`
+     is no longer dependent in the goal after application
+     of :tacn:`destruct`, it is erased (to avoid erasure, use parentheses, as
+     in :n:`destruct (@ident)`).
+
+   :n:`as @induction_ipat`
+      Provides names for the variables introduced
+      into the context.  The :token:`or_and_intropattern` must have the
+      form :n:`[ {*| {* @intropattern } } ]` with one :n:`{* @intropattern }`
+      for each constructor in the order in which the constructors are defined.
+      The items in
+      each :n:`{* @intropattern }` are used to name the variables created in
+      each new goal (one per constructor).  If there are not enough names,
+      Coq picks fresh names.
+
+   :n:`eqn : @naming_intropattern`
+      Generates a new hypothesis in each new subgoal that is an equality between
+      the term being case-analyzed and the associated constructor (applied to
+      its arguments).  The name of the new item may be specified in the
+      :n:`@naming_intropattern`.
+
+   :n:`with @bindings`  (in :n:`@one_term_with_bindings`)
+      Provides explicit instances for
+      the :term:`dependent premises <dependent premise>` of the type of
+      :token:`one_term`.
+
+   :n:`@occurrences`
+     Selects specific subterms of the goal and/or hypotheses to apply
+     the tactic to.  See :ref:`Occurrence clauses <occurrenceclauses>`.
+     If it occurs in the :n:`@induction_principle`, then
+     there can only be one :n:`induction_clause`, which can't have its
+     own :n:`occurrences` clause.
+
+   :n:`using @one_term {? with @bindings }`
+     Makes the tactic equivalent to :tacn:`induction` :n:`{+, @induction_clause }`.
+
+   .. tacn:: edestruct {+, @induction_clause } {? @induction_principle }
+
+      Equivalent to :n:`destruct @term` except that it does not
       fail if the instance of a dependent premises of the type
-      of :token:`term` is not inferable. Instead, the unresolved instances
+      of :token:`term` is not inferrable. Instead, the unresolved instances
       are left as existential variables to be inferred later, in the same way
-      as :tacn:`eapply` does.
+      they are in :tacn:`eapply`.
 
-   .. tacv:: destruct @term using @term {? with @bindings }
+.. tacn:: case {+, @induction_clause } {? @induction_principle }
 
-      This is synonym of :n:`induction @term using @term {? with @bindings }`.
+   .. todo PR: clear as mud? (below) the syntax accepts more parameters than elim
+      not sure what to say here
```

Comment:
Are all the left over `todo PR` still current?

Reply:
At line 206, for `case`, the description is pretty thin.  Maybe we can make it more descriptive?

Separately, shall I add `dependent` to the glossary as a synonym of `dependent product`?  (Note I used :term:`dependent <dependent product>` under `induction`, line 247.

The others are after line 250, so don't need to be addressed yet.

Reply:
>  Maybe we can make it more descriptive?

I don't know enough about this tactic to suggest what to add.

> Separately, shall I add `dependent` to the glossary as a synonym of `dependent product`? (Note I used :term:`dependent <dependent product>` under `induction`, line 247.

I didn't review that part yet (because I had misunderstood where to stop) but I don't find this sentence very clear and I'm not comfortable making these synonyms.